src/observer.ts

```diff
@@ -200,16 +217,36 @@ export class DataLayerObserver {
         handler.push(this.getOperator(optionSet));
       });
 
-      // optionally perform a final transformation
+      // optionally perform a final transformation if version is 1
       // useful if every rule needs the same operator run before the destination
-      if (beforeDestination) {
+      if (beforeDestination && (version === 1)) {
         const beforeOptions = Array.isArray(beforeDestination) ? beforeDestination : [beforeDestination];
         beforeOptions.forEach((operator) => handler.push(this.getOperator(operator)));
       }
 
-      // end with destination
-      const func = previewMode ? previewDestination : destination;
-      handler.push(new FunctionOperator({ name: 'function', func }));
+      if (fsApi) {
```

Comment:
Given these are operators, my skew is that we re-use the `OperatorFactory` and `this.getOperator` to instantiate operators.  I may have to get into the code to see if that creates unwanted complexity.